tests/src/python/providertestbase.py

```diff
@@ -120,6 +120,31 @@ def testGetFeaturesCompiled(self):
         except AttributeError:
             print 'Provider does not support compiling'
 
+    def testOrderBy(self):
+        request = QgsFeatureRequest().addOrderBy('cnt')
+        values = [f['cnt'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [-200, 100, 200, 300, 400])
+
+        request = QgsFeatureRequest().addOrderBy('cnt', False)
+        values = [f['cnt'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [400, 300, 200, 100, -200])
+
+        request = QgsFeatureRequest().addOrderBy('name')
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, ['Apple', 'Honey', 'Orange', 'Pear', NULL])
+
+        request = QgsFeatureRequest().addOrderBy('name', True, True)
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [NULL, 'Apple', 'Honey', 'Orange', 'Pear'])
+
+        request = QgsFeatureRequest().addOrderBy('name', False)
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [NULL, 'Pear', 'Orange', 'Honey', 'Apple'])
+
+        request = QgsFeatureRequest().addOrderBy('name', False, False)
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, ['Pear', 'Orange', 'Honey', 'Apple', NULL])
```

Comment:
Oh - one more test which would be good to see would be a test which relies on case-sensitivity of sorting. This would help identify if any providers have different behaviour with case sensitive sorting when compared to the reference QGIS implementation.